### DISCUSSION TITLE
Ideal filenames for effective parsing.

### DISCUSSION THREAD

**Discussion by tehguitarist** [comictagger/comictagger#615] [Q&A] [answered]
I know everyone seems to have a preference for how their files are named, but in terms of parsing them for metadata/tagging, is there an ideal? As an extension, is there a best place to add the title? This is to potentially help series that have two runs in the same year (same volume, different run - happened once on Uncanny Avengers I think) or TPB/HC which share a volume year as a run (lets say that a collected spider-man covering the first 1963 run is released in 2011, and there was a new run of spider-man in 2011 for example, but the title of  the TPB was "TPB" or "The Early Years" or whatever).

Currently I'm using:

{series} V{volume year} #{issue} ({year}) - The Amazing Spider-Man V1963 #001 (1963)

This seems to be fairly effective in most cases, though there's the odd miss that titles might help with. To add a title to the name, would it be best to add it at the end or before the year? And should there be a hyphen? I was thinking something like

{series} V{volume year} #{issue} ({year}) - {title}    -    The Amazing Spider-Man V1963 #001 (1963) - The Chameleon Strikes
or
{series} V{volume year} #{issue} - {title} ({year})    -    The Amazing Spider-Man V1963 #001 - The Chameleon Strikes (1963)

I probably prefer the latter, but I'm also cognisant of issues arising with filename parsing.

**Comment by lordwelch** [comment, marked as answer]
On the current alpha. All of those parse the same (except the first one not having the title). Currently the title is not used when searching for a comic. I have a bunch of tests for parsing different types of filenames here: https://github.com/comictagger/comictagger/blob/14ce8a759f629a22ee5a3e9561bb50e3a92af658/testing/filenames.py#L27 Currently all of them are parsed correctly as long as the issue number is present and has a `#` in front of it.

You can also see what exactly ComicTagger is able to parse out of a filename in the settings window as of [1.6.0-alpha.9](https://github.com/comictagger/comictagger/releases/tag/1.6.0-alpha.9)
![Screenshot_20240226_171316](https://github.com/comictagger/comictagger/assets/7547075/50521425-3068-4d81-b789-8849da75d09a)
